### PR TITLE
gh-107122: Add clear method to dbm.ndbm module

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-107122

----
:books: Documentation preview :books:: https://cpython-previews--107126.org.readthedocs.build/

### CODE REVIEW

**Comment by corona10** [comment]
cc @eendebakpt

**Comment by corona10** [comment]
## Refleak check

````
(.oss) ➜  cpython git:(gh-107122) ✗ ./python.exe -m test test_dbm -R 3:3
Raised RLIMIT_NOFILE: 256 -> 1024
0:00:00 load avg: 2.26 Run tests sequentially
0:00:00 load avg: 2.26 [1/1] test_dbm
beginning 6 repetitions
123456
......

== Tests result: SUCCESS ==

1 test OK.

Total duration: 258 ms
Tests result: SUCCESS
````

File: Modules/_dbmmodule.c [hunk 1]
```diff
@@ -414,6 +414,34 @@ _dbm_dbm_setdefault_impl(dbmobject *self, PyTypeObject *cls, const char *key,
...
+_dbm_dbm_clear_impl(dbmobject *self, PyTypeObject *cls)
+/*[clinic end generated code: output=8d126b9e1d01a434 input=43aa6ca1acb7f5f5]*/
+{
+    _dbm_state *state = PyType_GetModuleState(cls);
+    assert(state != NULL);
+    check_dbmobject_open(self, state->dbm_error);
+    for (datum key = dbm_firstkey(self->di_dbm); key.dptr; key = dbm_nextkey(self->di_dbm)) {
+        if (dbm_delete(self->di_dbm, key) < 0) {
+            dbm_clearerr(self->di_dbm);
+            PyErr_SetString(state->dbm_error, "cannot delete item from database");
```
Comment (vstinner) [inline Modules/_dbmmodule.c:442]: Do you have to update di_size in the error path? Maybe update the size at each iteration?

File: Lib/test/test_dbm.py [hunk 2]
```diff
@@ -155,6 +155,21 @@ def test_keys(self):
             self.assertNotIn(b'xxx', d)
             self.assertRaises(KeyError, lambda: d[b'xxx'])
 
+    def test_clear(self):
+        with dbm.open(_fname, 'c') as d:
+            self.assertEqual(d.keys(), [])
+            a = [(b'a', b'b'), (b'12345678910', b'019237410982340912840198242')]
```
Comment (vstinner) [inline Lib/test/test_dbm.py:161]: The second entry is a little bit complicated but it's just a remark, feel free to ignore 😁

File: Modules/_dbmmodule.c (hunk 1 above)
Comment (corona10) [reply Modules/_dbmmodule.c:442]: Done!

File: Lib/test/test_dbm.py (hunk 2 above)
Comment (corona10) [reply Lib/test/test_dbm.py:161]: Yeah it was just copied value from L148 :)

**Review by vstinner — APPROVED** [review]
LGTM, but please document the new feature in What's New In Python 3.13.

I would suggest "itemds", "key" and "value" names instead of a, k, v 😉

**Pull request merged**